build.c

```diff
@@ -182,16 +165,12 @@ buildadd(struct node *n)
 	e->flags &= ~FLAG_CYCLE;
 }
 
-static int
+static bool
```

Comment:
The error handling here is quite complicated so it's very possible that I broke it